sklearn/datasets/_base.py

```diff
@@ -659,7 +682,8 @@ def load_breast_cancer(*, return_X_y=False, as_frame=False):
         target_names=target_names,
         DESCR=fdescr,
         feature_names=feature_names,
-        filename=csv_filename,
+        filename=data_file_name,
+        file_module=DATA_MODULE
```

Comment:
To be compatible with `pyOxidizer` and `zipapp`, we can no longer pass around an absolute file path. Instead, we separately pass around `filename` (just the name of the file) and `file_module` that the file lives in.